exec.go

```diff
@@ -6,39 +6,40 @@ import (
 	"io"
 	"os"
 	"os/exec"
-	"syscall"
 
-	"github.com/creack/pty"
+	"github.com/charmbracelet/x/term"
+	"github.com/charmbracelet/x/xpty"
 )
 
 func executeCommand(ctx context.Context, args []string) ([]byte, error) {
-	cmd := exec.CommandContext(ctx, args[0], args[1:]...) //nolint: gosec
-	pty, err := runInPty(cmd)
+	width, height, err := term.GetSize(os.Stdout.Fd())
+	if err != nil {
+		width = 80
+		height = 24
+	}
+
+	pty, err := xpty.NewPty(width, height)
 	if err != nil {
 		return nil, err
 	}
-	defer pty.Close() //nolint: errcheck
+	defer func() {
+		_ = pty.Close()
+	}()
+
+	cmd := exec.CommandContext(ctx, args[0], args[1:]...) //nolint: gosec
+	if err := pty.Start(cmd); err != nil {
+		return nil, err
+	}
+
 	var out bytes.Buffer
 	var errorOut bytes.Buffer
 	go func() {
 		_, _ = io.Copy(&out, pty)
 		errorOut.Write(out.Bytes())
 	}()
 
-	err = cmd.Wait()
-	if err != nil {
+	if err := cmd.Wait(); err != nil {
```

Comment:
We probably want to use [`xpty.WaitProcess`](https://github.com/charmbracelet/x/blob/main/xpty/xpty.go#L75) for Windows

Reply:
This should only be used on Windows so we need to put it in a `runtime.GOOS == "windows"` block

Reply:
ohh, is it worth it making it do a simple cmd.wait if not windows?

Reply:
done https://github.com/charmbracelet/x/pull/273